02/app.js

```diff
@@ -1,32 +1,26 @@
 document.addEventListener('DOMContentLoaded', init);
 
 function init() {
-    const divList = document.querySelectorAll('div');
-    
-    setBorderColorAsync(divList[0], 'red', function() {
-        setBorderColorAsync(divList[1], 'blue', function() {
-            setBorderColorAsync(divList[2], 'green', function() {
-                console.log('finish');
-            });
-        });
-    });
+  const divList = document.querySelectorAll('div');
 
+  setBorderColorAsync(divList[0], 'red').then(() => {
+    setBorderColorAsync(divList[1], 'blue').then(() => {
+      setBorderColorAsync(divList[2], 'green')
+        .finally(console.log('Finish'))
+        .catch((err) => console.error(err));
+    });
+  });
```

Comment:
Ładniej byłoby tak:
```
setBorderColorAsync(divList[0], 'red')
    .then(() => setBorderColorAsync(divList[1], 'blue'))
    .then(() =>  setBorderColorAsync(divList[2], 'green'))
    .finally(() => console.log('Finish'))
    .catch((err) =>  console.error(err))
  });
```

Ponieważ, nie ma tylu zagnieżdżeń.
Pamiętaj, że to co zwracamy do `.then()` jest przekazywane do kolejnego `.then()`.